### PR TITLE
✨ v3 (feature): new mounting system

### PR DESCRIPTION
**Pull request by balcieren**
inspired by Express

- [x] fix sub in sub path bug
- [x] fix appList bug

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -724,6 +709,25 @@ func (app *App) All(path string, handlers ...Handler) Router {
 	return app
 }
 
+// The MountPath property contains one or more path patterns on which a sub-app was mounted.
+func (app *App) Mountpath() string {
```
Comment (efectn) [inline app.go:713]: I think `MountPath` would be better

File: app.go [hunk 2]
```diff
@@ -724,6 +709,25 @@ func (app *App) All(path string, handlers ...Handler) Router {
...
+	if app.mountpath == "" {
+		panic("mountpath cannot be used on parent app")
+	}
+
+	return utils.CopyString(app.mountpath)
+}
+
+// The mount event is fired on a sub-app, when it is mounted on a parent app. The parent app is passed to the callback function.
+func (app *App) OnMount(callback func(parent *App)) {
```
Comment (efectn) [inline app.go:722]: Add this into the hooks.

**Comment by ReneWerner87** [comment]
@balcieren there are some open tasks in the description, is the pull request ready or rather a draft ? can you please update it with the masters

**Comment by balcieren** [comment]
> @balcieren there are some open tasks in the description, is the pull request ready or rather a draft ? can you please update it with the masters

Actually I am working on v3-alpha branch. I have solved a lot of things about mounting, I am trying to best solution for mounting after that I am gonna fix this pr 👍

**Comment by efectn** [comment]
Can you just move mounting to Use since we won't fix sub in sub mounting in this PR. Also can you check the tests and review @balcieren

**Review by efectn — APPROVED** [review]

**Review by ReneWerner87 — APPROVED** [review]

**Comment by ReneWerner87** [comment]
thx @efectn and @balcieren

**Pull request merged**